src/renderer/component/publishForm/view.jsx

```diff
@@ -1,884 +1,574 @@
+  nameError: ?string,
+  isResolvingUri: boolean,
+  winningBidForClaimUri: number,
+  myClaimForUri: ?{
+    amount: number
+  },
+  licenseType: string,
+  otherLicenseDescription: ?string,
+  licenseUrl: ?string,
+  copyrightNotice: ?string,
+  uri: ?string,
+  bidError: ?string,
+  publishing: boolean,
+  publishError: ?string,
+  publishSuccess: boolean,
+  balance: number,
+  clearError: () => void,
+  navigate: (string) => void,
+  clearPublish: () => void,
+  clearFilePath: () => void,
+  resolveUri: (string) => void,
+}
 
-    if (!fetchingChannels) fetchChannelListMine();
+class PublishForm extends React.PureComponent<Props> {
+  static defaultProps = {
+    prefillClaim: {}
   }
 
-  handleSubmit() {
-    const { balance } = this.props;
-    const { bid } = this.state;
-
-    if (bid > balance) {
-      this.handlePublishError({ message: 'insufficient funds' });
-
-      return;
-    }
-
-    this.setState({
-      submitting: true,
-    });
-
-    const checkFields = this._requiredFields;
-    if (!this.myClaimExists()) {
-      checkFields.unshift('file');
-    }
-
-    let missingFieldFound = false;
-    for (const fieldName of checkFields) {
-      const field = this.refs[fieldName];
-      if (field) {
-        if (field.getValue() === '' || field.getValue() === false) {
-          field.showRequiredError();
-          if (!missingFieldFound) {
-            field.focus();
-            missingFieldFound = true;
-          }
-        } else {
-          field.clearError();
-        }
-      }
-    }
-
-    if (missingFieldFound) {
-      this.setState({
-        submitting: false,
-      });
-      return;
-    }
+  constructor(props: Props) {
+    super(props);
 
-    const metadata = {};
+    (this: any).handleFileChange = this.handleFileChange.bind(this);
+    (this: any).checkIsFormValid = this.checkIsFormValid.bind(this);
+    (this: any).renderFormErrors = this.renderFormErrors.bind(this);
+    (this: any).handlePublish = this.handlePublish.bind(this);
+    (this: any).handleCancelPublish = this.handleCancelPublish.bind(this);
+    (this: any).handleNameChange = this.handleNameChange.bind(this);
+    (this: any).handleChannelChange = this.handleChannelChange.bind(this);
+    (this: any).editExistingClaim = this.editExistingClaim.bind(this);
+    (this: any).getNewUri = this.getNewUri.bind(this);
+  }
+
+  handlePublish() {
+    const {
+      publish,
+      filePath,
+      bid,
+      title,
+      thumbnail,
+      description,
+      language,
+      nsfw,
+      channel,
+      licenseType,
+      licenseUrl,
+      otherLicenseDescription,
+      copyrightNotice,
+      name,
+      contentIsFree,
+      price
+    } = this.props;
 
-    for (const metaField of ['title', 'description', 'thumbnail', 'language']) {
-      const value = this.state[`meta_${metaField}`];
-      if (value) {
-        metadata[metaField] = value;
-      }
+    let publishingLicense;
+    switch (licenseType) {
+      case 'copyright':
```

Comment:
If these string values are shared across components, they should be defined as constant types similar to icons or actions.